partner_statement/report/current_statement.py

```diff
@@ -0,0 +1,160 @@
+# Copyright 2018 ForgeFlow, S.L. (https://www.forgeflow.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, models
+
+
+class CurrentStatement(models.AbstractModel):
+    """Model of Current Statement"""
+
+    _inherit = "statement.common"
+    _name = "report.partner_statement.current_statement"
+    _description = "Partner Current Statement"
+
+    def _display_lines_sql_q1(self, partners, date_end, account_type):
+        partners = tuple(partners)
+        return str(
+            self._cr.mogrify(
+                """
+            SELECT l.id, m.name AS move_id, l.partner_id, l.date, l.name,
+                            l.blocked, l.currency_id, l.company_id,
+            CASE WHEN l.ref IS NOT NULL
+                THEN l.ref
+                ELSE m.ref
+            END as ref,
+            CASE WHEN (l.currency_id is not null AND l.amount_currency > 0.0)
+                THEN avg(l.amount_currency)
+                ELSE avg(l.debit)
+            END as debit,
+            CASE WHEN (l.currency_id is not null AND l.amount_currency < 0.0)
+                THEN avg(l.amount_currency * (-1))
+                ELSE avg(l.credit)
+            END as credit,
+            CASE WHEN l.balance > 0.0
+                THEN round(l.balance - sum(coalesce(pd.amount, 0.0)),2)
```

Comment:
why apply round? if you have a good reason, then apply it to other reports too, but my opinion is not to apply it in the queries, as maybe you have a different rounding instead of 2. I think it should be applied somewhere in the python part of the code.